keys/default_rproc.pem

```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```

Comment:
Why do we need a key different from the `default_ta.pem`?

Reply:
> Why do we need a key different from the `default_ta.pem`?

Mainly to compartmentalize the keys used for the remoteproc firmware image and the Trusted Application.  For example, the remoteproc firmware key could be shared with a third party responsible for providing the coprocessor firmware.

Reply:
In that case, have you considered https://optee.readthedocs.io/en/latest/architecture/subkeys.html?

Reply:
> Thanks for the link. No i did not consider that. If I well understood your proposal is that the default_rproc key has to be considered as a subkey of the remoteproc TA , right?

No, the idea with subkeys is that you supply the chain of signed keys in the header. That way you can delegate signing of certain binaries without sharing the root key.

> So if that simplify the picture no issue to use the default_ta.pem'. Should I rename is to default_key.pem` in such case?

Sure, it might break signing of TAs in xtest, but that can be fixed also then.

Reply:
Related fixup added on top :
https://github.com/OP-TEE/optee_os/pull/6027/commits/fe6990cade8508af04721205e372755067d59ce9
https://github.com/OP-TEE/optee_os/pull/6027/commits/961a570631fe81ab54e45e9b1365551947062b08

Reply:
> CI Code style fail analysis:
> 
> * `Macro argument reuse 'x' - possible side-effects?`
>   check done, no side effect on macro
> * ` CHECK: Alignment should match open parenthesis`
>   i will add a fixup! to fix alignements

Alignement done in https://github.com/OP-TEE/optee_os/pull/6027/commits/84cd779e63dd55985581f2f23defb2335f6c3720

Reply:
> > Thanks for the link. No i did not consider that. If I well understood your proposal is that the default_rproc key has to be considered as a subkey of the remoteproc TA , right?
> 
> No, the idea with subkeys is that you supply the chain of signed keys in the header. That way you can delegate signing of certain binaries without sharing the root key.
> 
> > So if that simplify the picture no issue to use the default_ta.pem'. Should I rename is to default_key.pem` in such case?
> 
> Sure, it might break signing of TAs in xtest, but that can be fixed also then.

I created a PR to rename the key file name: https://github.com/OP-TEE/optee_os/pull/6048